dojo/unittests/tools/test_nuclei_parser.py

```diff
@@ -0,0 +1,122 @@
+from django.test import TestCase
+from dojo.tools.nuclei.parser import NucleiParser
+from dojo.models import Test
+
+
+class TestNucleiParser(TestCase):
+
+    def test_parse_no_findings(self):
+        testfile = open("dojo/unittests/scans/nuclei/no_findings.json")
+        parser = NucleiParser()
+        findings = parser.get_findings(testfile, Test())
+        self.assertEqual(0, len(findings))
+
+    def test_parse_many_findings(self):
+        testfile = open("dojo/unittests/scans/nuclei/many_findings.json")
+        parser = NucleiParser()
+        findings = parser.get_findings(testfile, Test())
+        self.assertEqual(9, len(findings))
```

Comment:
Please add also cleaning: 

```python
for finding in findings:
    for endpoint in finding.unsaved_endpoints:
        endpoint.clean()
```

https://github.com/DefectDojo/cookiecutter-scanner-parser/blob/main/temp-%7B%7B%20cookiecutter.tool_directory_name%20%7D%7D/unittests/tools/test_%7B%7B%20cookiecutter.tool_directory_name%20%7D%7D_parser.py#L20-L22

Reply:
added

Reply:
it is good to add it before all `self.assertEqual("...", finding.unsaved_endpoints[0]....)` because `.clean()` can change content of fields in `Endpoint`

Reply:
Done